.github/workflows/cno-ci.yml

```diff
@@ -0,0 +1,131 @@
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.GOPATH }}/src/k8s.io/kubernetes/
+          key: k8s-go-2-${{ env.K8S_VERSION }}
+
+      - name: Set up Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ env.GO_VERSION }}
+        id: go
+        if: steps.cache-k8s.outputs.cache-hit != 'true'
+
+      - name: Build and install Kubernetes
+        if: steps.cache-k8s.outputs.cache-hit != 'true'
+        run: |
+          set -x
+          rm -rf $GOPATH/src/k8s.io/kubernetes
+          git clone --single-branch --branch $K8S_VERSION https://github.com/kubernetes/kubernetes.git $GOPATH/src/k8s.io/kubernetes/
+          pushd $GOPATH/src/k8s.io/kubernetes/
+          make WHAT="test/e2e/e2e.test vendor/github.com/onsi/ginkgo/ginkgo cmd/kubectl"
+          rm -rf .git
+
+  e2e:
+    name: e2e
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+          - shard-n
+          - shard-np
+          - shard-s
+          - shard-other
+        ha:
+          - enabled: "false"
+            name: "noHA"
+    needs: k8s
+    env:
+      JOB_NAME: "${{ matrix.target }}-${{ matrix.ha.name }}"
+      KIND_HA: "${{ matrix.ha.enabled }}"
+    steps:
+
+      - name: Free up disk space
+        run: sudo eatmydata apt-get remove --auto-remove -y aspnetcore-* dotnet-* libmono-* mono-* msbuild php-* php7* ghc-* zulu-*
+
+      - name: Set up Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ env.GO_VERSION }}
+        id: go
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+
+      - name: Set up environment
+        run: |
+          export GOPATH=$(go env GOPATH)
+          echo "::set-env name=GOPATH::$GOPATH"
+          export PATH=$GOPATH/bin:$PATH
+          echo "::add-path::$GOPATH/bin"
+
+      - name: Restore Kubernetes from cache
+        id: cache-k8s
+        uses: actions/cache@v2
+        with:
+          path: "${{ env.GOPATH }}/src/k8s.io/kubernetes/"
+          key: k8s-go-2-${{ env.K8S_VERSION }}
+
+      # Re-build if kube wasn't in the cache due to
+      # https://github.com/actions/cache/issues/107#issuecomment-598188250
+      # https://github.com/actions/cache/issues/208#issuecomment-596664572
+      - name: Build and install Kubernetes
+        if: steps.cache-k8s.outputs.cache-hit != 'true'
+        run: |
+          set -x
+          rm -rf $GOPATH/src/k8s.io/kubernetes/
+          git clone --single-branch --branch $K8S_VERSION https://github.com/kubernetes/kubernetes.git $GOPATH/src/k8s.io/kubernetes/
+          pushd $GOPATH/src/k8s.io/kubernetes/
+          make WHAT="test/e2e/e2e.test vendor/github.com/onsi/ginkgo/ginkgo cmd/kubectl"
+          rm -rf .git
+
+      - name: CNO KIND setup
+        run: |
+          make -C test install-cno-kind
+
+      - name: Run Tests
+        run: |
+          make -C test ${{ matrix.target }}
```

Comment:
If I read it correctly, the current version of the script creates a single node cluster, you need at least 2 worker and 1 master to run the e2e.test against it. This adds it openshift/cluster-network-operator#663

Reply:
@aojea I skipped e2e for this PR for the reason you pointed out and was going to start working out how to get multi-node for CNO KIND. Im pumped you already did. E2e will still break without hybrid overlay enabled. I will ping you and maybe we can get it added on your PR.

Reply:
feel free to take my PR and consolidate it with yours, so we can be sure everything works perfectly